Dockerfile

```diff
@@ -1,11 +1,38 @@
-# Use ChainGuard's glibc-dynamic image as the base image. More information about this image can be found at https://www.chainguard.dev/chainguard-images
-FROM cgr.dev/chainguard/glibc-dynamic
+# More information about this image can be found at https://www.chainguard.dev/chainguard-images
+FROM cgr.dev/chainguard/wolfi-base
+
+RUN apk add libstdc++
 
 # Declare a build-time argument for the target architecture
 ARG TARGETARCH
 
 # Add the binary file 'surreal' from the specified path on the host machine to the root directory in the container
 ADD $TARGETARCH/surreal /
 
-# Set the entry point for the container to be the 'surreal' binary
-ENTRYPOINT ["/surreal"]
+# Add the 'surreal' user to the container
+RUN adduser -D surreal
```

Comment:
Small suggestion, can we just add explicit user, group and volume directory inside the container?

```
# Create a surreal user and group.
RUN addgroup surreal && \
    adduser -S -G surreal surreal

# The /surreal/data directory used by SurrealDB to store state. 
RUN mkdir -p /surreal/data && \
    chown -R surreal:surreal /surreal

VOLUME /surreal/data
```


and in the entry point just 

if [ "$1" = 'surreal' ]; then
  # Allow setting up the user and group via environment variables
  if [ -z "$SURREAL_UID" ]; then
    SURREAL_UID="$(id -u surreal)"
  fi

  if [ -z "$SURREAL_GID" ]; then
    SURREAL_GID="$(id -g surreal)"
  fi

  # If the data or config dirs are bind mounted, then chown them.
  if [ "$(stat -c %u "$SURREAL_DATA_DIR")" != "${SURREAL_UID}" ]; then
    chown ${SURREAL_UID}:${SURREAL_GID} "$SURREAL_DATA_DIR"
  fi

  set -- su-exec ${SURREAL_UID}:${SURREAL_GID} "$@"
fi

Reply:
I was just thinking if it would be helpful to provide these as env variables (UID/GID). Please ignore if unnecessary